ick/runner.py

```diff
@@ -59,33 +62,71 @@ def iter_rule_impl(self) -> Iterable[BaseRule]:
             i = get_impl(rule)(rule, self.rtc)
             yield i
 
-    def test_rules(self) -> None:
-        with ThreadPoolExecutor() as tpe, Progress() as progress:
-            outstanding = {}
-            prepare_key = progress.add_task("Prepare", total=None)
+    def test_rules(self) -> int:
+        """
+        Returns an exit code (0 on success)
+        """
+        with ThreadPoolExecutor() as tpe:
+            print("[dim]testing...[/dim]")
+            # It's about this point I realize that yes, I am writing a whole
+            # test runner and that's not the business I want to be in.  Oh
+            # well...
+            buffered_output = []
+
+            i = 0
             for rule_instance, names in self.iter_tests():
+                outstanding = {}
+                print(f"  [bold]{rule_instance.rule_config.qualname}[/bold]: ", end="")
                 rule_instance.prepare()
```

Comment:
No need to prepare if there are no test names?

Reply:
I think trying to prepare regardless gets us better finding of bugs in rules?

Reply:
This is still in the serial portion, so the user will see it through normal python means.  I hope to red-diff most of this code soon and use the normal run path for tests.